### ISSUE TITLE
About how to proceed with alerts

### ISSUE THREAD

**Issue by aljesusg** [ManageIQ/manageiq-providers-hawkular#99]
Now when you create a alert of a type in Miq, hawkular is not creating the trigger for each Datasource. I have little knowledge about hawkular and I didn't know this. So here is the problem and some proposal.

## Datasource

MiQ is looking for
`MI~R~[0b84c5546795/Local DMR~~]~MT~Datasource Pool Metrics~In Use Count`

Hawkular is providing metrics in
`MI~R~[92da52a9b841/Local DMR~/subsystem=datasources/data-source=ExampleDS]~MT~Datasource Pool Metrics~In Use Count`

**/subsystem=datasources/data-source=ExampleDS**

So If yo wanna trigger the alert you need to push data 

```ruby
client.metrics.gauges.push_data("MI~R~[0b84c5546795/Local DMR~~]~MT~Datasource Pool Metrics~In Use Count",{:value=>37})
```

## Mesagging

MiQ is looking
`MI~R~[0b84c5546795/Local~~]~MT~JMS Topic Metrics~Subscription Count`

Hawkular is providing in
`MI~R~[92da52a9b841/Local~/subsystem=messaging-activemq/server=default/jms-topic=HawkularAlertsActionsTopic]~MT~JMS Topic Metrics~Subscription Count"`

**/subsystem=messaging-activemq/server=default/jms-topic=HawkularAlertsActionsTopic**

With Web Sessions there isn't any problem.

## Problems

- MiQ provider is not creating the trigger for each Datasource/Mesagging type in MWServer

## Proposal Solutions

1.  fix de manageiq-providers-hawkular to create the trigger for each DataSource in the MiddlewareServer selected
2. As @lucasponce  told us in https://github.com/ManageIQ/manageiq-providers-hawkular/pull/88 make changes in MiQ UI to select Datasource type, Mesagging...

cc @abonas , @mtho11, @jdoyleoss

**Comment by jshaughn** [comment]
The comments in #88 are correct, and the options above make sense.  My question is whether this is going to be resolved for TP3.  Because if not then we should wait for Prometheus migration and then proceed.

As for the approach, I'd probably suggest option 2, being able to assign to specific DS or Messaging entities.  The alternative could result in a lot of unnecessary triggers and possibly unwanted alerts.

**Comment by aljesusg** [comment]
I am working in the second option I think is the best too.

**Comment by jdoyleoss** [comment]
I agree the second option is the better option. Let's pursue that.

~john

**Comment by abonas** [comment]
I agree with the second option. this is NOT for gaprindashvili though.

**Comment by abonas** [comment]
@aljesusg / @jdoyleoss please log an appropriate JMAN4 as this is a functional enhancement
